### DISCUSSION TITLE
Is it possible to search with case insensitive?

### DISCUSSION THREAD

**Discussion by maximdx** [ast-grep/ast-grep#1142] [Q&A] [answered]
I hope to use it to search a large code base, but I cannot remember the exact cases.
Is this support in ast-grep?

**Comment by HerringtonDarkholme** [comment]
hi, would you like to provide more context and use case examples? thanks

**Comment by maximdx** [comment]
Sure. I am very new to ast-grep.
I plan to use ast-grep to do a code pattern search on my working folder.
For example:
`sg -p 'mysearchpattern' -l java ./myworkingfolder`

I hope to use the pattern 'mysearchpattern' to march multiple cases like: mySearchPattern, or MySearchPattern or even MYSeArCHPATTern (I know this one is unlikely to happen). But at the moment, it only matches exact the input pattern.

**Reply by HerringtonDarkholme** [reply, marked as answer]
You can use [YAML rule](https://ast-grep.github.io/guide/rule-config.html) and [regex](https://ast-grep.github.io/guide/rule-config/atomic-rule.html#regex) rule
